nixos/lib/default.nix

```diff
@@ -0,0 +1,33 @@
+let
+  # The warning is in a top-level let binding so it is only printed once.
+  minimalModulesWarning = warn "lib.nixos.evalModules is experimental and subject to change. See nixos/lib/default.nix" null;
+  inherit (nonExtendedLib) warn;
+  nonExtendedLib = import ../../lib;
+in
+{ # Optional. Allows an extended `lib` to be used instead of the regular Nixpkgs lib.
+  lib ? nonExtendedLib,
+
+  # Feature flags allow you to opt in to unfinished code. These may change some
+  # behavior or disable warnings.
+  featureFlags ? {},
+
+  # This file itself is rather new, so we accept unknown parameters to be forward
+  # compatible. This is generally not recommended, because typos go undetected.
+  ...
+}:
+let
+  seqIfNot = cond: if !cond then builtins.seq else a: b: b;
+  # If cond, force `a` before returning any attr
+  seqIfNotAttrs = cond: a: lib.mapAttrs (_: v: seqIfNot cond a v);
```

Comment:
this is a bit confusing. at first we read it as `apply seq if a is not attrs` and were surprised by what it does. maybe `seqAttrsIfNot` is clearer. (we'd even drop the `Not` in make them `seqIf` and `seqAttrsIf`)

Reply:
`?`'s strange operator priority compared to `.` and function application biased my choice here, but that was entirely unnecessary.